### PR TITLE
Adding integration test for PYTHON-486

### PR DESCRIPTION
**Pull request by GregBestland**
Fixing some bad test documentation, and adding integration test for PYTHON-486.

### CODE REVIEW

File: tests/integration/cqlengine/connections/test_connection.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+        @test_category object_mapper
+        """
+
+        connection.set_session(self.session1)
+        sync_table(TestConnectModel)
+        TCM1 = TestConnectModel.create(id=1, keyspace=self.keyspace1)
+        connection.set_session(self.session2)
+        sync_table(TestConnectModel)
+        TCM2 = TestConnectModel.create(id=1, keyspace=self.keyspace2)
```
Comment (aholmberg) [inline tests/integration/cqlengine/connections/test_connection.py]: If we're going to use count below, I think we would need to change the id to make sure we're not just overwriting in the same ks/table.
Alternatively, use if-not-exists when creating, or manually inspect cluster metadata to make sure the sync_table targets the expected keyspace.

Comment (GregBestland) [inline tests/integration/cqlengine/connections/test_connection.py]: I could be missing something here, but I think we could catch such a instance in our asserts below. That was the purpose of the separately named keyspace values in the table. If such an over-write occurred the asserts below fail. As the key space value would not match what was expected. You would end up fetching the same keyspace value from the model twice which wouldn't pass muster.

Comment (aholmberg) [inline tests/integration/cqlengine/connections/test_connection.py]: ok. I guess I was just confused.

File: tests/integration/cqlengine/connections/test_connection.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        cls.setup_cluster.shutdown()
+
+    def setUp(self):
+        self.c = Cluster(protocol_version=PROTOCOL_VERSION)
+        self.session1 = self.c.connect(keyspace=self.keyspace1)
+        self.session1.row_factory = dict_factory
+        self.session2 = self.c.connect(keyspace=self.keyspace2)
+        self.session2.row_factory = dict_factory
+
+    def test_connection_session_switch(self):
```
Comment (aholmberg) [inline tests/integration/cqlengine/connections/test_connection.py]: I think we might leak a few sessions here: session1, session2, and the one that is originally setup in cqlengine.connection. Should we shut those down?

Comment (GregBestland) [inline tests/integration/cqlengine/connections/test_connection.py]: Agreed.

**Comment by GregBestland** [comment]
Issues should now be addressed.

**Pull request merged**